### PR TITLE
Test Session::addMessageAfterRedirect; refs #950

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: tests/SessionTest.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      $warn_msg = 'There was a warning. Be carefull.';
+      $info_msg = 'All goes well. Or not... Who knows ;)';
+
+      $this->assertNull($_SESSION['MESSAGE_AFTER_REDIRECT']);
+
+      //test add message in cron mode
+      $_SESSION['glpicronuserrunning'] = 'phpunit_cron';
+      Session::addMessageAfterRedirect($err_msg, false, ERROR);
+
+      $this->assertNull($_SESSION['MESSAGE_AFTER_REDIRECT']);
```
Comment (orthagh) [inline tests/SessionTest.php]: AssertNotNull, no ? You added a message above.

File: tests/SessionTest.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      $expected = [
+        ERROR   => [$err_msg],
+        WARNING => [$warn_msg],
+        INFO    => [$info_msg]
+      ];
+
+      $this->expectOutputRegex('/' . str_replace('.', '\.', $err_msg)  . '/');
+      $this->expectOutputRegex('/' . str_replace('.', '\.', $warn_msg)  . '/');
+      $this->expectOutputRegex('/' . str_replace(['.', ')'], ['\.', '\)'], $info_msg)  . '/');
+      Html::displayMessageAfterRedirect();
```
Comment (orthagh) [inline tests/SessionTest.php]: is the expect functions and the display need to be inverted ?
You do the same couple further.
Perhaps i miss something but expectOutputRegex tests the php output the display function send, no ?

File: tests/SessionTest.php (hunk 1 above)
Comment (trasher) [inline tests/SessionTest.php]: Nope, we indeed add a message, but is Session::isCron() is true, anything will be added in the session ;)

File: tests/SessionTest.php (hunk 2 above)
Comment (trasher) [inline tests/SessionTest.php]: Yes, it tests the output of the method. As for the order, I've just followed phpunit's documentation (https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.output) - does not sounds illogic to me to expect before and output after - this is not a variable.

File: tests/SessionTest.php (hunk 1 above)
Comment (orthagh) [inline tests/SessionTest.php]: ok, this seems a good explanation

File: tests/SessionTest.php (hunk 2 above)
Comment (orthagh) [inline tests/SessionTest.php]: also ok

File: tests/SessionTest.php (hunk 1 above)
Comment (ddurieux) [inline tests/SessionTest.php]: Perhaps add mode comment about this part of text in the file for the future ;)

Comment (trasher) [inline tests/SessionTest.php]: Done ;)

**Comment by trasher** [comment]
Thanks!

**Pull request merged**